src/rust/storage/seg/src/hashtable/mod.rs

```diff
@@ -410,6 +410,47 @@ impl HashTable {
         }
     }
 
+    pub(crate) fn is_item_at(&self, key: &[u8], seg: NonZeroU32, offset: u64) -> bool {
+        let hash = self.hash(key);
+        let tag = tag_from_hash(hash);
+        let bucket_id = hash & self.mask;
+
+        let mut bucket = &self.data[bucket_id as usize];
+        let chain_len = chain_len(bucket.data[0]);
+        let mut chain_idx = 0;
+
+        loop {
```

Comment:
why is this not `for chain_idx in 0..chain_len {` ? It'd be a little more concise that way I think.

Reply:
Excellent question! This should probably be changed in every place we're scanning through the hash buckets. This was just copy past from another function that does something similar. 

It could probably be expressed as some sort of iterator itself... but for now I'll cleanup the loops

Reply:
Fixed.

Reply:
#414 for converting to an iterator in the future.